lib/http3/common.c

```diff
@@ -470,6 +470,9 @@ static void process_packets(h2o_http3_ctx_t *ctx, quicly_address_t *destaddr, qu
     if (conn == NULL) {
         /* Initial or 0-RTT packet, use 4-tuple to match the thread and the connection */
         assert(packets[0].cid.dest.might_be_client_generated);
+        /* we can't handle packets of that length */
+        if (packets[0].cid.src.len > QUICLY_MAX_CID_LEN_V1)
+            return;
```

Comment:
How about moving the check above, to where we check the version? I.e.,

```c
if (QUICLY_PACKET_IS_LONG_HEADER(packets[0].octets.base[0])) {
    if (packets[0].version != QUICLY_PROTOCOL_VERSION) {
        send VN...
    } else if (packets[0].cid.src.len > QUICLY_MAX_CID_LEN_V1) {
        return;
    }
}
```

I agree that it would be possible to delay the check until as late as being proposed by the PR, I think we might prefer simplicity here, because I do not think we'd have any performance benefit in practice.

Reply:
That makes sense, @kazuho . How does 5241fd1